galasa-ui/src/app/layout.tsx

```diff
@@ -13,32 +13,36 @@ import { FeatureFlagProvider } from '@/contexts/FeatureFlagContext';
 import { cookies } from 'next/headers';
 import FeatureFlagCookies from '@/utils/featureFlagCookies';
 import { getLocale } from 'next-intl/server';
+import { ThemeProvider } from '@/contexts/ThemeContext';
 
 export const dynamic = "force-dynamic";
+type ThemeType = "white" | "g100"; 
 
-export default async function RootLayout({children,}: {children: React.ReactNode}) {
-  // Ensure that the incoming `locale` is valid
+export default async function RootLayout({ children }: { children: React.ReactNode }) {
   const locale = await getLocale();
- 
+
   const galasaServiceName = process.env.GALASA_SERVICE_NAME?.trim() || "Galasa Service";
   const featureFlagsCookie = cookies().get(FeatureFlagCookies.FEATURE_FLAGS)?.value;
+  const themeCookie = cookies().get('preferred-theme')?.value;
+  const initialTheme: ThemeType = themeCookie === 'g100' ? 'g100' : 'white'; // safe fallback
 
   return (
-    <html lang={locale}>
+    <html lang={locale} data-carbon-theme={initialTheme}>
       <head>
         <title>{galasaServiceName}</title>
         <meta name="description" content="Galasa Ecosystem Web UI" />
       </head>
       <body>
         <NextIntlClientProvider>
           <FeatureFlagProvider initialFlags={featureFlagsCookie}>
-            <PageHeader galasaServiceName={galasaServiceName} />
-            {children}
-            <Footer serviceHealthyPromise={getServiceHealthStatus()} clientVersionPromise={getClientApiVersion()}/>
+            <ThemeProvider initialTheme={initialTheme}>
```

Comment:
Do we need to pass in an initial theme here? Or could the initial theme be set by this theme provider?

That way, localStorage can be used to get the theme instead of a cookie (assuming the ThemeProvider is for client-side use).